SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/interpolate/ArrayInterpolator.java

```diff
@@ -703,19 +703,21 @@ public WeakPathInfo(final ProofPath path) {
 					final Term storeIndex = getIndexFromStore(storeTerm);
 					final ApplicationTerm indexDiseq =
 							mDisequalities.get(new SymmetricPair<Term>(storeIndex, mPathIndex));
-					mTail.closeAPath(mHead, boundaryTerm, stepOcc);
 					if (indexDiseq != null) {
 						final Occurrence indexDiseqOcc = mInterpolator.getLiteralInfo(indexDiseq);
 						final Occurrence intersectOcc = stepOcc.intersect(indexDiseqOcc);
 
+						mTail.closeAPath(mHead, boundaryTerm, stepOcc);
 						mTail.closeAPath(mHead, boundaryTerm, intersectOcc);
```

Comment:
I'm still not convinced that this is correct.  Do we really need to open/close paths if the index disequality belong to another partition?  It should be enough to keep the paths rolling and only add the index disequality.